microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+
+* **v8.compile**: this event shows the compilation time of your content script.
+* **v8.run**: this event indicates the running time of the compiled script.
+
+
+<!-- ====================================================================== -->
+## Only add content script code that's needed for the functionality of your extension
+
+Your extension's content script runs in the context of the webpage. To minimize the impact of your content script on that webpage, make sure to only add, in your content script, the minimum amount of code that your extension needs to run in the context of the webpage. Audit the code in your content script and remove legacy frameworks, tools, libraries, or other code that aren't needed by your content script to run in Microsoft Edge.
+
+You can make use of lazy loading and code splitting techniques to minimize the amount of code that runs in your content script:
+
+* Lazy loading is the process of loading is the process of loading your code only when it's needed, based on the user actions, the page content, or the extension logic.
+* Code splitting is the process of dividing your code into smaller chunks, or modules, which can be loaded separately or on demand.
+
+If your extension is small enough, you won't need a build tool to split your code. If your extension is larger, and if your code is more complex to manage, use a build to split your code into smaller chunks. Build tools can help you organize your code into logical units, which can be loaded on demand. For example, you can use webpack to split your code into _entry points_ and _dynamic imports_:
+
+* Entry points are loaded on every page load.
+* Dynamic imports are only loaded on demand, for example when the user interacts with the webpage or with the extension's UI:
+
+  ```javascript
+  // When the user clicks on the page.
+  document.addEventListener("click", async () => {
+    // Dynamically load the code that's needed to handle the click event.
+    const module = await import("chunk.js");
+    // Do something with the newly loaded module code.
+  });
+  ```
+
+
+<!-- ====================================================================== -->
+## Only load your content script in required pages and frames
+
+Your extension might not need to run on every webpage that the user visits. To reduce the amount of code that runs when webpages load, configure your extension to load content scripts only on the pages and frames where they are needed.
+
+To configure the pages and frames where your content scripts are loaded, define URL patterns in your extension manifest file by using the `matches` property in the `content_scripts` section. To learn more, see [Inject scripts](https://developer.chrome.com/docs/extensions/develop/concepts/content-scripts#functionality) in the Chrome Extensions documentation.
```

Comment:
```suggestion
To configure the pages and frames where your content scripts are loaded, define URL patterns in your extension manifest file by using the `matches` property in the `content_scripts` section. To learn more, see [Inject scripts](https://developer.chrome.com/docs/extensions/develop/concepts/content-scripts#functionality) in the Chrome Extensions documentation.
```
link text "Inject scripts" appears to contradict anchor "functionality".  try: see [heading] in _article title_.